### ISSUE TITLE
"declare global" prevents accessing parent scope (no-undef)

### ISSUE THREAD

**Issue by JamesHenry** [typescript-eslint/typescript-eslint#21]
This issue was initially reported here: https://github.com/eslint/typescript-eslint-parser/issues/579

---

**What version of TypeScript are you using?**
3.1.3

**What version of `typescript-eslint-parser` are you using?**
21.0.2

**What code were you trying to parse?**

```ts
import React from 'react';

declare global {
  interface Global {
    __r: typeof React;
  }
}

// To complement `global.__r = React;`
```

**What did you expect to happen?**
No lint errors

**What happened?**
>   6:18  error  'React' is not defined  no-undef

typescript-eslint-parser/analyze-scope.js:Referencer.visitGlobalAugmentation seems to replace the current scope with the global one, which makes accessing `React` in the above example 
incorrectly say it's not defined. 

@mysticatea (via #540): what case does visitGlobalAugmentation solve?

**Comment by ifiokjr** [comment]
@JamesHenry I'm running into the same error. It only happens a few times in my codebase so it's easy enough to ignore, but I'm wondering if there's a workaround or if there are any plans to update the rule.

**Comment by bradzacher** [comment]
Recommendation is to turn off the rule, as the typescript compiler itself should handle everything that the rule covers.

Otherwise - no plans right now. It involves adding scope analysis support, which is a huge piece of work none of the core maintainers have time to work on right now.

**Comment by bradzacher** [comment]
Merging into #1856
